nav2_world_model/include/nav2_world_model/world_representation.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_WORLD_MODEL__WORLD_REPRESENTATION_HPP_
+#define NAV2_WORLD_MODEL__WORLD_REPRESENTATION_HPP_
+
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_msgs/srv/get_costmap.hpp"
+#include "nav2_msgs/srv/process_region.hpp"
+
+namespace nav2_world_model
+{
+
+using nav2_msgs::srv::GetCostmap;
+using nav2_msgs::srv::ProcessRegion;
+
+class WorldRepresentation
+{
+public:
+  explicit WorldRepresentation(const std::string & name, rclcpp::Node::SharedPtr & node)
+  : name_(name), node_(node)
+  {
+  }
+
+  WorldRepresentation() = delete;
+
+  virtual GetCostmap::Response getCostmap(const GetCostmap::Request & request) = 0;
```

Comment:
These different functions should be abstracted from the world model in some way, so that it can be extended with arbitrary capabilities int the future. As I see how you are using it, it seems it could be a map of std::function objects. You could do something like `worldRepresentation["GetCostmap"]()` in the client code or something.

Reply:
I think this will change significantly once we define the layers and plugins from #565.